### PR TITLE
fetch-mc-role-from-aws-cluster

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/giantswarm/issues/24359

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
```
Comment (fiunchinho) [inline CHANGELOG.md:10]:
```suggestion
### Changed

```

File: controllers/awscluster_controller.go [hunk 2]
```diff
@@ -153,9 +152,19 @@ func (r *AWSClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return reconcile.Result{}, err
 	}
 
+	// Fetch AWSClusterRole from the cluster.
+	awsManagementClusterRoleIdentity := &capa.AWSClusterRoleIdentity{}
+	err = r.Get(ctx, client.ObjectKey{Name: managementAWSCluster.Spec.IdentityRef.Name}, awsManagementClusterRoleIdentity)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
```
Comment (fiunchinho) [inline controllers/awscluster_controller.go:159]: I don't think we should ignore the error when the identity doesn't exist. If we have an `AWSCluster` using a non existing identity, we should return errors.

**Review by fiunchinho — CHANGES_REQUESTED** [review]
I have one small request.

**Comment by fiunchinho** [comment]
Wouldn't we need to do the same when deleting a cluster?

**Comment by calvix** [comment]
> Wouldn't we need to do the same when deleting a cluster?

its just the pre-init phase the loops for normal reconciliation or deletion are below this code, so it will apply to both

**Pull request merged**